modules/network/src/test/java/org/apache/ignite/internal/network/serialization/ClassDescriptorParserTest.java

```diff
@@ -0,0 +1,355 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.ignite.internal.network.serialization.DefaultClassDescriptors.DefaultType;
+import org.apache.ignite.lang.IgniteException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Set of tests for the class descriptor parser.
+ */
+public class ClassDescriptorParserTest {
+    /**
+     * Descriptor parser context.
+     */
+    private final ClassDescriptorParserContext context = new ClassDescriptorParserContext();
+
+    /**
+     * Descriptor parser.
+     */
+    private final ClassDescriptorParser parser = new ClassDescriptorParser(context);
+
+    static class SerializableClass implements Serializable {
+        private static final long serialVersionUID = 0L;
+    }
+
+    @Test
+    public void testSerializable() {
+        ClassDescriptor descriptor = parser.parse(SerializableClass.class);
+
+        assertEquals(SerializationType.SERIALIZABLE, descriptor.serializationType());
+        assertEquals(0, descriptor.fields().size());
+    }
+
+    static class SerializableOverride implements Serializable {
+        private static final long serialVersionUID = 0L;
+
+        private void writeObject(ObjectOutputStream out) throws IOException {
+            // No-op.
+        }
+
+        private void readObject(ObjectInputStream in)
+                throws IOException, ClassNotFoundException {
+            // No-op.
+        }
+
+        private void readObjectNoData() throws ObjectStreamException {
+            // No-op.
+        }
+    }
+
+    @Test
+    public void testSerializableOverride() {
+        ClassDescriptor descriptor = parser.parse(SerializableOverride.class);
+
+        assertEquals(SerializationType.SERIALIZABLE_OVERRIDE, descriptor.serializationType());
+        assertEquals(0, descriptor.fields().size());
+    }
+
+    static class SerializableOverrideWriteReplace implements Serializable {
```

Comment:
I guess you should also test cases when signatures of these methods do not match the required signatures